### ISSUE TITLE
startWith doesn't start with given value for streams created from futures that resolve immediately

### ISSUE THREAD

**Issue by mhnowak** [ReactiveX/rxdart#453]
Hey,
Today I've upgraded to 0.24.0 dart and my tests failed. So I started digging thru my code to find the issue. I basically have a bit more complicated code that works somehow similar to the demo I've made: 
```dart
void main() {
  Future<bool> asyncFunc() async => false;

  Stream.fromFuture(asyncFunc()).startWith(true).listen(print);
}
```
To my surprise it resolves into 
```
false
true
```
being printed. I've noticed the behaviour of `.startWith` has been changed with introduction of 0.24.0. Is this change also intended? After checking [RxMarbles](https://rxmarbles.com/#startWith) I believe it's just a bug.

For now I downgraded to 0.23.1 version, where same code prints: 
```
true
false
```
Also think it's worth mentioning that it works fine if we delay the `asyncFunc` with something like `Future.delayed`. The issue occurs when a future resolves immediately.

**Comment by frankpepermans** [comment]
This should be resolved already in the master branch, would you mind trying it out?
We'll do a 0.25.0 release soon

**Comment by mhnowak** [comment]
Seems to work as expected on master. Thanks, looking forward to the new version ;)

**Comment by hoc081098** [comment]
This has already been fixed in version 0.24.1.
